pulsar-client/src/main/java/org/apache/pulsar/client/impl/TableViewImpl.java

```diff
@@ -230,16 +230,26 @@ private void handleMessage(Message<T> msg) {
         }
     }
 
-    private CompletableFuture<Reader<T>> readAllExistingMessages(Reader<T> reader) {
+    @Override
+    public CompletableFuture<Void> refreshAsync() {
+        return reader.thenCompose(this::readAllExistingMessages);
```

Comment:
`readAllExistingMessages` will call `readTailMessages` after the future is completed. It will cause many threads running `readTailMessages` if `refreshAsync` is called many times.

Reply:
Good catch!